crates/nu-utils/standard_library/dirs.nu

```diff
@@ -0,0 +1,83 @@
+# Maintain a list of working directories and navigates them
+
+# the directory stack
+export-env {
+    let-env DIRS_POSITION = 0
+    let-env DIRS_LIST = [($env.PWD | path expand)]
+}
+
+# Add one or more directories to the list.
+# PWD becomes first of the newly added directories.
+export def-env "add" [
+    ...paths: string    # directory or directories to add to working list
+    ] {
+        mut abspaths = []
+        for p in $paths {
+            let exp = ($p | path expand)
+            if ($exp | path type) != 'dir' {
+                let span = (metadata $p).span
+                error make {msg: "not a directory", label: {text: "not a directory", start: $span.start, end: $span.end } }
+            }
+        $abspaths = ($abspaths | append $exp)
+
+        }
+        let-env DIRS_LIST = ($env.DIRS_LIST | insert ($env.DIRS_POSITION + 1) $abspaths | flatten)
+        let-env DIRS_POSITION = $env.DIRS_POSITION + 1
+
+    _fetch 0
+}
+
+# Advance to the next directory in the list or wrap to beginning.
+export def-env "next" [
+    N:int = 1   # number of positions to move.
+] {
+    _fetch $N    
+}
+
+# Back up to the previous directory or wrap to the end.
+export def-env "prev" [
+    N:int = 1   # number of positions to move.
+] {
+    _fetch (-1 * $N)    
+}
+
+# Drop the current directory from the list, if it's not the only one.
+# PWD becomes the next working directory
+export def-env "drop" [] {
+    if ($env.DIRS_LIST | length) > 1 {
+        let-env DIRS_LIST = (
+            ($env.DIRS_LIST | take $env.DIRS_POSITION) 
+            | append ($env.DIRS_LIST | skip ($env.DIRS_POSITION + 1))
+        )
+    }
+
+    _fetch 0
+}
+
+# Display current working directories.
+export def-env "show" [] {
+    mut out = []
+    for $p in ($env.DIRS_LIST | enumerate) {
+        $out = ($out | append [
+            [active, path]; 
+            [($p.index == $env.DIRS_POSITION), $p.item]
+        ])
+    }
+
+    $out
+}
+
+# fetch item helper
+def-env  _fetch [
+    offset: int,    # signed change to position
+] {
+    # nushell 'mod' operator is really 'remainder', can return negative values.
+    # see: https://stackoverflow.com/questions/13683563/whats-the-difference-between-mod-and-remainder    
+    let pos = ($env.DIRS_POSITION 
+                + $offset 
+                + ($env.DIRS_LIST | length)
+            ) mod ($env.DIRS_LIST | length)
+    let-env DIRS_POSITION = $pos
+
+    cd ($env.DIRS_LIST | get $pos )
+}
```

Comment:
ah, so that's how it's done!  
I was looking for something in github that would apply the changes from the PR...

And what tool are you running that does this formatting and uncovering these lints?   I'm running vscode with TheNuProjectContributors.vscode-nushell-lang  v1.0.0, is there some better combination?

Reply:
> ah, so that's how it's done! I was looking for something in github that would apply the changes from the PR...
> 
yup, that's the easiest way i know to apply a patch from *GitHub* :wink: 

> And what tool are you running that does this formatting and uncovering these lints? I'm running vscode with TheNuProjectContributors.vscode-nushell-lang v1.0.0, is there some better combination?
>
the tool is called "my brain and my hands" :eyes: :laughing: 

on really, there is no such thing for now, but that's a thing i want to work on some time very soon :yum:

Reply:
I'm finding that at least some of my excess whitespace is coming from vscode expanding tab chars to blanks.  I am *not* going to change that setting and risk reigniting the clone wars!
I've turned on white space rendering so I can see better, and eagerly await the addition of a standard nushell formatting tool!

Reply:
indeed. someone needs to finish `nufmt` already https://github.com/fdncred/nufmt 🤣

Reply:
cool :yum: 

i have trailing whitespaces hightlighting enabled in `neovim`, that's why i saw yours :wink: